src/Cmd/Add.cc

```diff
@@ -124,8 +127,14 @@ addDependencyToManifest(
     }
   }
 
-  auto data = toml::parse(getManifestPath());
-  auto& deps = toml::find<toml::table>(data, "dependencies");
+  // Keep the order of the tables.
+  auto data = toml::parse<toml::ordered_type_config>(getManifestPath());
+
+  // Check if the dependencies table exists, if not create it.
+  if (data["dependencies"].is_empty()) {
+    data["dependencies"] = toml::table{};
+  }
+  auto& deps = data["dependencies"];
```

Comment:
We should handle the case where `data["dependencies"]` is not a table, shouldn't we?

Reply:
Great, thanks for your clarification.